linkerd/app/core/src/handle_time.rs

```diff
@@ -259,7 +259,7 @@ impl Shared {
         if counter.clones.fetch_sub(1, Ordering::Release) == 1 {
             let elapsed = t0.elapsed();
 
-            let mut hist = match self.histogram.lock() {
+            let hist = match self.histogram.lock() {
```

Comment:
since we no longer need to mutate the histogram, should we consider using a rwlock? (could be a follow-up)

Reply:
@hawkw sounds reasonable but yeah, let's save that for a followup